edk2toolext/tests/test_self_describing_environment.py

```diff
@@ -52,8 +61,31 @@ def test_override_path_env(self):
         tree.create_path_env("testing_corebuild2", var_name="jokes", scope=custom_scope,
                              extra_data={"override_id": "testing_corebuild"})
         build_env, shell_env = self_describing_environment.BootstrapEnvironment(self.workspace, scopes)
-        print(build_env.paths)
-        print(tree.get_workspace())
+        self.assertEqual(len(build_env.paths), 1)
+
+    def test_multiple_override_path_env(self):
+        ''' checks the SDE descriptor override system can handle multiple overrides'''
+        custom_scope = "global"
+        scopes = (custom_scope,)
+        tree = uefi_tree(self.workspace, create_platform=False)
+        tree.create_path_env("testing_corebuild", var_name="hey", dir_path="test1", scope=custom_scope)
+        tree.create_path_env("testing_corebuild2", var_name="jokes", scope=custom_scope,
+                             extra_data={"override_id": "testing_corebuild"})
+        tree.create_path_env("testing_corebuild3", var_name="laughs", scope=custom_scope,
+                             extra_data={"override_id": "testing_corebuild"})
+        # we should get warnings but still have two overrides
```

Comment:
So both descriptors are included in system state and the overridden file is not.

Reply:
Is there an example of where that's useful or desirable?
If sufficient example is provided, can we really hammer on that idea in the docs, because that's not what I would expect from reading them, especially after the "more than one id is an error" I think I tacitly apply that idea to overried_id.

Reply:
we talked about this....but this doesn't seem right. 
What do you mean "both descriptors are included in system state"?  

How can you have two that are included?  which one wins?  
I think this is an error.  

Now maybe i could agree that the descriptor with the highest priority scope wins.  But if they are both at same scope then its an error.  If different scopes that I would suggest that only the highest priority scope would show up as "active" in system state.  The overridden descriptor would show up as overridden and the lower scope descriptor would show up as "ignored"  or something like that. 

Scope order/priority vs override id is an interesting intersection i don't think we have talked about in the docs.

Reply:
So there isn't an ignored or overridden state outside the SDE. There is just in or out. So the simplest and most user friendly thing to do is just throw an error. This means less undefined behavior.